dolfyn/tools/misc.py

```diff
@@ -44,20 +48,20 @@ def detrend(arr, axis=-1, in_place=False):
     sz = np.ones(arr.ndim, dtype=int)
     sz[axis] = arr.shape[axis]
     x = np.arange(sz[axis], dtype=np.float_).reshape(sz)
-    x -= x.mean(axis=axis, keepdims=True)
-    arr -= arr.mean(axis=axis, keepdims=True)
-    b = (x * arr).mean(axis=axis, keepdims=True) / (x ** 2).mean(axis=axis,
-                                                                 keepdims=True)
+    x -= np.nanmean(x, axis=axis, keepdims=True)
+    arr -= np.nanmean(arr, axis=axis, keepdims=True)
+    b = np.nanmean((x * arr), axis=axis, keepdims=True)/ \
+        np.nanmean((x ** 2), axis=axis, keepdims=True)
     arr -= b * x
     return arr
 
 
-def group(bl, min_length=0):
```

Comment:
This one is also useful to expose to the API.

Reply:
Done